Source/data/iterators.hpp

```diff
@@ -1,234 +1,408 @@
 #pragma once
 
 #include <charconv>
+#include <ostream>
+
+#include <expected.hpp>
 
 #include "parser.hpp"
 
 namespace devilution {
-class FieldsInRecordRange {
+
+class Field {
 	GetFieldResult *state_;
 	const char *const end_;
+	const unsigned row_;
+	const unsigned column_;
 
 public:
-	FieldsInRecordRange(GetFieldResult *state, const char *end)
+	Field(GetFieldResult *state, const char *end, unsigned row, unsigned column)
 	    : state_(state)
 	    , end_(end)
+	    , row_(row)
+	    , column_(column)
 	{
 	}
 
-	class Iterator {
-		GetFieldResult *state_;
-		const char *const end_;
-
-	public:
-		using iterator_category = std::input_iterator_tag;
-		using value_type = std::string_view;
-
-		Iterator()
-		    : state_(nullptr)
-		    , end_(nullptr)
-		{
+	/**
+	 * @brief Returns a view of the current field
+	 *
+	 * This method scans the current field if this is the first value access since the last
+	 * advance. If you expect the field to contain a numeric value then calling parseInt first
+	 * is more efficient, but calling the methods in either order is supported.
+	 * @return The current field value (may be an empty string) or a zero length string_view
+	 */
+	[[nodiscard]] std::string_view value()
+	{
+		if (state_->status == GetFieldResult::Status::ReadyToRead) {
+			*state_ = GetNextField(state_->next, end_);
 		}
+		return state_->value;
+	}
 
-		Iterator(GetFieldResult *state, const char *end)
-		    : state_(state)
-		    , end_(end)
-		{
-			state_->status = GetFieldResult::Status::ReadyToRead;
-		}
+	/**
+	 * Convenience function to let Field instances be used like other single-value STL containers
+	 */
+	[[nodiscard]] std::string_view operator*()
+	{
+		return this->value();
+	}
 
-		[[nodiscard]] bool operator==(const Iterator &rhs) const
-		{
-			if (state_ == nullptr && rhs.state_ == nullptr)
-				return true;
+	/**
+	 * Allows using a field transparently as a string_view
+	 */
+	[[nodiscard]] operator std::string_view()
+	{
+		return this->value();
+	}

```

Comment:
Implicit conversions are a feature of C++ best avoided most of the time. Let's remove this conversion and be explicit when we need a `string_view` (it's just a `*` after all). https://google.github.io/styleguide/cppguide.html#Implicit_Conversions

Reply:
removed the operator==(std::string_view) definition as well since it had a similar effect.